internal/admission/defaulting_test.go

```diff
@@ -111,3 +111,77 @@ func TestTimeout(t *testing.T) {
 		require.InDelta(t, timeout.Seconds(), time.Since(start).Seconds(), 0.1, "timeout duration is not respected")
 	})
 }
+
+func TestStrictMode(t *testing.T) {
+	tests := []struct {
+		name        string
+		strictMode  bool
+		wantedLabel string
+	}{
+		{
+			name:        "Strict mode on",
+			strictMode:  true,
+			wantedLabel: pkg.ValidationStatusReject,
+		},
+		{
+			name:        "Strict mode off",
+			strictMode:  false,
+			wantedLabel: pkg.ValidationStatusPending,
+		},
+	}
+
+	logger := zap.NewNop()
```

Comment:
Can we unify the initialization of this machine to functions? As it's similar to the first use case

Reply:
Done as suggested.